lib/core/cmap/connection.js

```diff
@@ -0,0 +1,211 @@
+'use strict';
+
+const EventEmitter = require('events');
+const MessageStream = require('./message_stream');
+const MongoError = require('../error').MongoError;
+const MongoWriteConcernError = require('../error').MongoWriteConcernError;
+const wp = require('../wireprotocol');
+const apm = require('../connection/apm');
+const updateSessionFromResponse = require('../sessions').updateSessionFromResponse;
+
+const kStream = Symbol('stream');
+const kQueue = Symbol('queue');
+const kMessageStream = Symbol('messageStream');
+
+class Connection extends EventEmitter {
+  constructor(stream, options) {
+    super(options);
+
+    this.id = `${stream.address().address}:${stream.address().port}`;
```

Comment:
Is `stream` a socket?

Reply:
yeah, I just don't want us to forget that we are expecting it to be a socket. I think a `TODO` will suffice for now.

Reply:
added `streamIdentifier` to generate a uuidV4 if `address()` isn't available